### PR TITLE
cluster-estimation-integration-redo-redo

### PR DESCRIPTION
**Pull request by Aleksa-M**

### CODE REVIEW

File: main_2024.py [hunk 1]
```diff
@@ -111,6 +112,15 @@ def main() -> int:
...
         GEOLOCATION_CAMERA_ORIENTATION_ROLL = config["geolocation"]["camera_orientation_roll"]
+
+        MIN_ACTIVATION_THRESHOLD = config["cluster_estimation"]["min_activation_threshold"]
+        MIN_NEW_POINTS_TO_RUN = config["cluster_estimation"]["min_new_points_to_run"]
+        RANDOM_STATE = config["cluster_estimation"]["random_state"]
+
+        MIN_ACTIVATION_THRESHOLD = config["cluster_estimation"]["min_activation_threshold"]
+        MIN_NEW_POINTS_TO_RUN = config["cluster_estimation"]["min_new_points_to_run"]
+        RANDOM_STATE = config["cluster_estimation"]["random_state"]
+
```
Comment (mgupta27) [inline main_2024.py:123]: This is a repeat of the code block above.

File: main_2024.py [hunk 2]
```diff
@@ -345,6 +375,19 @@ def main() -> int:
 
     worker_managers.append(geolocation_manager)
 
+    result, cluster_estimation_manager = worker_manager.WorkerManager.create(
+        worker_properties=cluster_estimation_worker_properties,
+        local_logger=main_logger,
+    )
+    if not result:
+        main_logger.error("Failed to create manager for Flight Interface", True)
```
Comment (mgupta27) [inline main_2024.py:383]: `Failed to create manager for Cluster Estimation`

File: main_2024.py [hunk 3]
```diff
@@ -357,7 +400,7 @@ def main() -> int:
                 return -1
 
         try:
-            geolocation_data = geolocation_to_main_queue.queue.get_nowait()
+            geolocation_data = geolocation_to_cluster_estimation_queue.queue.get_nowait()
```
Comment (mgupta27) [inline main_2024.py:403]: We do not want to log geolocation in the main logger as it has its own logging now. You can get rid of the entire try block.

File: main_2024.py [hunk 4]
```diff
@@ -375,6 +418,16 @@ def main() -> int:
                     "geolocation confidence: " + str(detection_world.confidence), True
                 )
 
+        try:
+            cluster_estimations = cluster_estimation_to_main_queue.queue.get_nowait()
+        except queue.Empty:
+            cluster_estimations = None
+        if cluster_estimations is not None:
```
Comment (mgupta27) [inline main_2024.py:425]: Add a space above.

**Review by mgupta27 — COMMENTED** [review]
Reviewed.

File: main_2024.py (hunk 4 above)
Comment (Aleksa-M) [reply main_2024.py:425]: pylint is telling me to not add a blank line below cluster_estimations = None

File: config.yaml [hunk 5]
```diff
@@ -34,3 +34,8 @@ geolocation:
     camera_orientation_yaw: 0.0
     camera_orientation_pitch: -1.57079632679
     camera_orientation_roll: 0.0
+
+cluster_estimation:
+    min_activation_threshold: 25
+    min_new_points_to_run: 5
+    random_state: 0
```
Comment (maxlou05) [inline config.yaml:41]: Please add a newline to the end here

**Review by maxlou05 — COMMENTED** [review]
reviewed

File: main_2024.py [hunk 6]
```diff
@@ -277,6 +287,22 @@ def main() -> int:
...
+        count=1,
+        target=cluster_estimation_worker.cluster_estimation_worker,
+        work_arguments=(MIN_ACTIVATION_THRESHOLD, MIN_NEW_POINTS_TO_RUN, RANDOM_STATE),
+        input_queues=[geolocation_to_cluster_estimation_queue],
+        output_queues=[cluster_estimation_to_main_queue],
+        controller=controller,
+        local_logger=main_logger,
+    )
+    if not result:
+        main_logger.error("Failed to create arguments for Video Input", True)
```
Comment (mgupta27) [inline main_2024.py:300]: Sorry I missed this in the first round of review. Change the error to `Failed to create arguments for Cluster Estimation`.

File: main_2024.py [hunk 7]
```diff
@@ -357,24 +396,16 @@ def main() -> int:
...
-                main_logger.debug("geolocation label: " + str(detection_world.label), True)
-                main_logger.debug(
-                    "geolocation confidence: " + str(detection_world.confidence), True
-                )
-
+            cluster_estimations = None
+
+        if cluster_estimations is not None:
+            for cluster in cluster_estimations:
+                main_logger.debug("Cluser in world: True")
```
Comment (mgupta27) [inline main_2024.py:405]: Cluster is mispelt in this line. Also the debug function is missing the second parameter which should be set to `True`.

File: main_2024.py [hunk 8]
```diff
@@ -357,24 +396,16 @@ def main() -> int:
...
+                main_logger.debug("Cluster location x: " + str(cluster.location_x))
+                main_logger.debug("Cluster location y: " + str(cluster.location_y))
+                main_logger.debug("Cluster spherical variance: " + str(cluster.spherical_variance))
```
Comment (mgupta27) [inline main_2024.py:408]: These are all missing the second parameter of debug set to `True`.

**Review by mgupta27 — COMMENTED** [review]
Reviewed.

File: main_2024.py (hunk 7 above)
Comment (maxlou05) [reply main_2024.py:405]: It's true by default

File: main_2024.py (hunk 8 above)
Comment (maxlou05) [reply main_2024.py:408]: True by default

**Pull request merged**